py/desiutil/install.py

```diff
@@ -573,17 +573,16 @@ def set_install_dir(self):
         :class:`str`
             The directory selected for installation.
         """
-        try:
-            self.nersc = os.environ['NERSC_HOST']
-        except KeyError:
-            self.nersc = None
-        if self.options.root is None or not os.path.isdir(self.options.root):
-            if self.nersc is not None:
-                self.options.root = self.default_nersc_dir()
-            else:
-                message = "Root install directory is missing or not set."
-                self.log.critical(message)
-                raise DesiInstallException(message)
+        self.nersc = os.getenv('NERSC_HOST') #- will be None if not set
```

Comment:
Is this particular change really necessary?  I really hate `os.getenv()` because it never throws an exception, so it's much harder to tell when a variable is undefined.

Reply:
OK.  I reverted the `os.getenv()` to use `try/except` around `os.environ` again, while noting that simply catching the exception and setting the variable to `None` is exactly the same thing that `os.getenv` does if the environment variable isn't defined.  I agree that `os.getenv()` is a landmine if the code assumes that the variable is defined and would choke if silently given a `None`, but in this case `None` is the correct fallback, so `os.getenv` seemed simpler.  (And now I've spent 7 lines of text explaining why I prefered 1 line of code over 4... :)

I think you were only commenting on the `os.getenv()` line, but just to be clear: the rest of the changes here are important, because they are fixing the problem where desiInstall was silently falling back to the default NERSC location if the user-specified root directory didn't exist.

Reply:
I realize both old and new have the same result. So why change it?

Cosmetic changes like this distract from the real changes. I've heard that before in some of my own pull requests.

Although there apparently is no active unit test on this bit of code, the try: clause would potentially be something that is deliberately triggered in a unit test.

Reply:
Noted and agreed about mixing cosmetic changes distracting from the substantive changes.  In this case it was the result of several iterations of sorting out what this code should do and various attempts to find the best order of if/then try/except for what should be checked in what order, and it landed back there in the N-1 iteration.  I have reverted it back your preferred method.